Userland/Libraries/LibGfx/ImageDecoder.cpp

```diff
@@ -19,60 +20,81 @@
     auto size = bytes.size();
+    OwnPtr<ImageDecoderPlugin> plugin;
 
-    auto plugin = [](auto* data, auto size) -> OwnPtr<ImageDecoderPlugin> {
-        OwnPtr<ImageDecoderPlugin> plugin;
+    plugin = make<PNGImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PNGImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<GIFImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<GIFImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<BMPImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<BMPImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PBMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PBMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PGMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PGMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PPMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PPMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<ICOImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<ICOImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<JPGImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<JPGImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<DDSImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<DDSImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<QOIImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<QOIImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    return {};
+}
 
-        plugin = make<TGAImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+RefPtr<ImageDecoder> ImageDecoder::try_create_for_raw_bytes(ReadonlyBytes bytes)
+{
+    OwnPtr<ImageDecoderPlugin> plugin = probe_and_sniff_for_appropriate_plugin(bytes);
+    if (!plugin)
+        return {};
+    return adopt_ref_if_nonnull(new (nothrow) ImageDecoder(plugin.release_nonnull()));
+}
+RefPtr<ImageDecoder> ImageDecoder::try_create_for_mapped_file_path(StringView path, ReadonlyBytes bytes)
```

Comment:
Just a thought: do we need to pass the entire path to the ImageDecoder, or can we make do with just the extension (or filename)?

Reply:
I rather make it so we only create a `LexicalPath` once at this function instead of requiring everyone to do that.